substrate/pkg/controller/substrate/cluster/instance.go

```diff
@@ -78,6 +78,7 @@ func (i *Instance) Create(ctx context.Context, substrate *v1alpha1.Substrate) (r
 		if strings.Contains(aws.StringValue(err.ErrorMessage), "Invalid IAM Instance Profile name") {
 			return reconcile.Result{Requeue: true}, nil
 		}
+		return reconcile.Result{}, fmt.Errorf("creating fleet", aws.StringValue(err.ErrorMessage))
```

Comment:
If there are other/more errors we want to return an error, but if we know its `Invalid IAM Instance Profile name` we want to requeue.
I guess its fine, because if the first error is not `Invalid IAM Instance Profile name` we will return an error.